hibernate-reactive-core/src/main/java/org/hibernate/reactive/context/impl/VertxContext.java

```diff
@@ -16,17 +16,21 @@
  *
  * @author Gavin King
  */
-public class VertxContext implements Context, ServiceRegistryAwareService {
+public final class VertxContext implements Context, ServiceRegistryAwareService {
```

Comment:
Do we really want to stop people subclassing this? I would have thought that was useful.

Reply:
When it's a fairly simple `Service` implementation, I'd encourage people to implement their own from scratch by implementing the base interface (`Context` in this case).

This is also more flexible as you might not want to make it `ServiceRegistryAwareService` for example, which is implied by this particular implementation.

Reply:
Well it seemed to me pretty likely that you might just want to override `execute()` without overriding any of the other methods.

Reply:
As a user, I find it pretty annoying having to re-implement a whole class when I want to change only one method of the existing one.

Reply:
If anything, let's clarify that the recommended approach is to implement the interface but let's not make life harder for developers that know what they are doing.

Reply:
well fine I'll remove the `final` ... but this is an `impl` class, it seems intended for people to back off. And I just don't expect many users to have any need to change how this works.